packages/plugin-js-packages/src/lib/config.ts

```diff
@@ -0,0 +1,83 @@
+import { z } from 'zod';
+import { IssueSeverity, issueSeveritySchema } from '@code-pushup/models';
+
+const packageCommandSchema = z.enum(['audit', 'outdated']);
+export type PackageCommand = z.infer<typeof packageCommandSchema>;
+
+const packageManagerSchema = z.enum(['npm', 'yarn', 'yarn-berry', 'pnpm']);
+export type PackageManager = z.infer<typeof packageManagerSchema>;
+
+const packageAuditLevelSchema = z.enum([
+  'info',
+  'low',
+  'moderate',
+  'high',
+  'critical',
+]);
+export type PackageAuditLevel = z.infer<typeof packageAuditLevelSchema>;
+
+const defaultAuditLevelMapping: Record<PackageAuditLevel, IssueSeverity> = {
+  critical: 'error',
+  high: 'error',
+  moderate: 'warning',
+  low: 'warning',
+  info: 'info',
+};
+
+export function fillAuditLevelMapping(
+  mapping: Partial<Record<PackageAuditLevel, IssueSeverity>>,
+): Record<PackageAuditLevel, IssueSeverity> {
+  return {
+    critical: mapping.critical ?? defaultAuditLevelMapping.critical,
+    high: mapping.high ?? defaultAuditLevelMapping.high,
+    moderate: mapping.moderate ?? defaultAuditLevelMapping.moderate,
+    low: mapping.low ?? defaultAuditLevelMapping.low,
+    info: mapping.info ?? defaultAuditLevelMapping.info,
+  };
+}
+
+// TODO how?
+// export function objectKeys<T extends object>(obj: T): (keyof T)[] {
+//   return Object.keys(obj) as (keyof T)[];
+// }
+
+// function newFillAuditLevelMapping(
+//   mapping: Partial<Record<PackageAuditLevel, IssueSeverity>>,
+// ): Record<PackageAuditLevel, IssueSeverity> {
+//   return Object.fromEntries(
+//     objectKeys(defaultAuditLevelMapping).map<
+//       [PackageAuditLevel, IssueSeverity]
+//     >(auditLevel => [
+//       auditLevel,
+//       mapping[auditLevel] ?? defaultAuditLevelMapping[auditLevel],
+//     ]),
+//   );
+// }
```

Comment:
:raised_eyebrow: :question:

Reply:
I attempted to fill the record dynamically and failed. But unless the audit levels change a lot, even the explicit mapping is fine. WDYT?

Reply:
The audit levels change hardly ever, I believe.